File: 프로그래머스/3/49191. 순위/순위.js
function solution(n, results) {
    const table = Array.from({length: n+1}, () => {
        return Array.from({length: n+1}, () => 0);
    });
    
    results.forEach(([win, lose]) => {
        table[win][lose] = 1;
        table[lose][win] = -1;
    });
    
    let is_update = true;
    while (is_update) {
        is_update = false;
        for (let i = 1; i <= n; i++) {
            for (let j = 1; j <= n; j++) {
                if (table[i][j] === 0) continue;
                table[j].forEach((value, index) => {
                    if (value === table[i][j] && table[i][index] === 0) {
                        table[i][index] = value;
                        is_update = true;
                    }
                });
            }
        }
    }
    
    table.forEach(arr => {
        const count = arr.filter(v => v === 0).length;
        if (count > 2) n--;
    });
    
    return n + 1;
}

File: 프로그래머스/3/49191. 순위/순위.py
def solution(n, results):
    table = [[0 for _ in range(n+1)] for __ in range(n+1)]
    
    for win, lose in results:
        table[win][lose] = 1
        table[lose][win] = -1
    
    is_update = True
    
    while is_update:
        is_update = False
        for i in range(1, n+1):
            for j in range(1, n+1):
                if table[i][j] == 0: continue
                for index, value in enumerate(table[j]):
                    if value == table[i][j] and table[i][index] == 0:
                        table[i][index] = value
                        is_update = True
                        
    for arr in table:
        if arr.count(0) > 2:
            n -= 1
    
    return n + 1
